samples/interactive-tutorials/events/import-user-events-big-query.js

```diff
@@ -14,16 +14,20 @@
 
 'use strict';
 
-async function main(datasetId) {
+async function main(datasetId, impersonatedPrincipal) {
   // [START retail_import_user_events_big_query]
 
   // Imports the Google Cloud client library.
   const {UserEventServiceClient} = require('@google-cloud/retail').v2;
 
+  // check if auth with impersonated client.
+  const {impersonatedAuthClient} = require('../setup/setup-cleanup');
+  const {opts, projectId} = await impersonatedAuthClient(impersonatedPrincipal);
+
   // Instantiates a client.
-  const retailClient = new UserEventServiceClient();
+  const retailClient = new UserEventServiceClient(opts);
 
-  const projectId = await retailClient.getProjectId();
+  // const projectId = await retailClient.getProjectId();
```

Comment:
`  const {opts, projectId} = await impersonatedAuthClient(impersonatedPrincipal);`
because we grab the value from the util function already, as we shall override the `auth` object, the retailClient object won't have projectId included, that's why we shall return the `projectId` from the util function directly.